prisma/schema.prisma

```diff
@@ -31,3 +33,75 @@ model Session {
   createdAt DateTime @default(now()) @db.Timestamptz(6)
   updatedAt DateTime @updatedAt @db.Timestamptz(6)
 }
+
+model ArtifactTemplate {
+  id             String          @id @default(uuid()) @db.Uuid
+  title          String
+  fieldTemplates FieldTemplate[]
+  artifacts      Artifact[]
+  user           User?           @relation(fields: [userId], references: [id])
+  userId         String?         @db.Uuid
+  visibility     Visibility
+  createdAt      DateTime        @default(now()) @db.Timestamptz(6)
+  updatedAt      DateTime        @updatedAt @db.Timestamptz(6)
+}
+
+model FieldTemplate {
+  id                 String           @id @default(uuid()) @db.Uuid
+  order              Int              @unique
+  aiPrompt           String?
+  type               FieldType
+  description        String?
+  placeholder        String?
+  required           Boolean          @default(false)
+  artifactTemplate   ArtifactTemplate @relation(fields: [artifactTemplateId], references: [id])
+  artifactTemplateId String           @db.Uuid
+  fields             Field[]
+  createdAt          DateTime         @default(now()) @db.Timestamptz(6)
+  updatedAt          DateTime         @updatedAt @db.Timestamptz(6)
+}
+
+model Artifact {
+  id                 String           @id @default(uuid()) @db.Uuid
+  fields             Field[]
+  artifactTemplate   ArtifactTemplate @relation(fields: [artifactTemplateId], references: [id])
+  artifactTemplateId String           @db.Uuid
+  user               User             @relation(fields: [userId], references: [id])
+  userId             String           @db.Uuid
+  createdAt          DateTime         @default(now()) @db.Timestamptz(6)
+  updatedAt          DateTime         @updatedAt @db.Timestamptz(6)
+}
+
+model Field {
+  id              String        @id @default(uuid()) @db.Uuid
+  order           Int           @unique
+  text            String?
+  image           Image?
+  fieldTemplate   FieldTemplate @relation(fields: [fieldTemplateId], references: [id])
+  fieldTemplateId String        @db.Uuid
+  artifact        Artifact      @relation(fields: [artifactId], references: [id])
+  artifactId      String        @db.Uuid
+  createdAt       DateTime      @default(now()) @db.Timestamptz(6)
+  updatedAt       DateTime      @updatedAt @db.Timestamptz(6)
+}
+
+model Image {
+  id        String   @id @default(uuid()) @db.Uuid
+  title     String
+  url       String
+  field     Field    @relation(fields: [fieldId], references: [id])
+  fieldId   String   @unique @db.Uuid
+  createdAt DateTime @default(now()) @db.Timestamptz(6)
+  updatedAt DateTime @updatedAt @db.Timestamptz(6)
+}
+
+enum FieldType {
+  Image
+  Text
+  WebLink
```

Comment:
I believe we need both singular and plural for both `WebLink` (`WebLinks`) and `Image` (`Images`)

Reply:
There are a few others that we had on our whiteboard but this is a fine start